engines/experiment/pkg/request/request.go

```diff
@@ -82,3 +127,31 @@ func getValueFromJSONPayload(body []byte, key string) (string, error) {
 			"Field %s can not be parsed as string value, unsupported type: %s", key, typez.String())
 	}
 }
+
+// UPIVariablesToStringMap convert slice of upi Variables into map of string
```

Comment:
Clarification, this looks like a common enough function to be exposed as a helper method from UPI repository directly, or would this not be so commonly used, by other components of CaraML?

Reply:
Good suggestion. But so far it's only used by Turing. Will evaluate it if it's necessary to add it in the UPI library.